components/brave_wallet/common/fil_address.cc

```diff
@@ -0,0 +1,180 @@
+// static
+FilAddress FilAddress::FromAddress(const std::string& address) {
+  if (address.size() != kAddressSizeBLS &&
+      address.size() != kAddressSizeSecp256K)
+    return FilAddress();
+
+  auto protocol = ToProtocol(address[1]);
+  if (!protocol)
+    return FilAddress();
+
+  std::string network{address[0]};
+  if (!IsValidNetwork(network))
+    return FilAddress();
+
+  std::string payload{
+      base32::Base32Decode(base::ToUpperASCII(address.substr(2)))};
+  if (payload.empty())
+    return FilAddress();
+
+  std::string key{payload.substr(0, payload.size() - kChecksumSize)};
+  std::vector<uint8_t> public_key(key.begin(), key.end());
+  return FilAddress::FromPublicKey(public_key, protocol.value(), network);
+}
+
+// Creates FilAddress from SECP256K uncompressed public key
+// with specified protocol  and network
+// https://spec.filecoin.io/appendix/address/#section-appendix.address.string
+// static
+FilAddress FilAddress::FromUncompressedPublicKey(
+    const std::vector<uint8_t>& uncompressed_public_key,
+    mojom::FilecoinAddressProtocol protocol,
+    const std::string& network) {
+  if (protocol == mojom::FilecoinAddressProtocol::BLS)
+    return FilAddress();
+  if (uncompressed_public_key.empty())
+    return FilAddress();
+  auto public_key = BlakeHash(uncompressed_public_key, kPublicKeySizeSecp256K);
+  if (!public_key || public_key->empty())
+    return FilAddress();
+  return FromPublicKey(*public_key, protocol, network);
+}
+
+// Creates FilAddress from SECP256K or BLS public key
+// with specified protocol  and network
+// https://spec.filecoin.io/appendix/address/#section-appendix.address.string
+// static
+FilAddress FilAddress::FromPublicKey(const std::vector<uint8_t>& public_key,
+                                     mojom::FilecoinAddressProtocol protocol,
+                                     const std::string& network) {
+  if (!IsValidNetwork(network))
+    return FilAddress();
+  if (protocol == mojom::FilecoinAddressProtocol::SECP256K1) {
+    if (public_key.size() != kPublicKeySizeSecp256K)
+      return FilAddress();
+  } else if (protocol == mojom::FilecoinAddressProtocol::BLS) {
+    if (public_key.size() != kPublicKeySizeBLS)
+      return FilAddress();
+  }
+  return FilAddress(public_key, protocol, network);
+}
+
+// static
+bool FilAddress::IsValidAddress(const std::string& address) {
+  return !address.empty() &&
+         FilAddress::FromAddress(address).ToChecksumAddress() == address;
+}
+
+// https://spec.filecoin.io/appendix/address/#section-appendix.address.string
+// |------------|----------|---------|----------|
+// |  network   | protocol | payload | checksum |
+// |------------|----------|---------|----------|
+// | 'f' or 't' |  1 byte  | n bytes | 4 bytes  |
+std::string FilAddress::ToChecksumAddress() const {
+  if (bytes_.empty())
+    return std::string();
+  std::vector<uint8_t> payload(bytes_);
+  std::vector<uint8_t> checksumPayload(bytes_);
```

Comment:
Please don't use camel case for local variables in c++, should be checksum_payload for example.